cardano-db-sync/src/Cardano/DbSync.hs

```diff
@@ -363,12 +363,18 @@ chainSyncClient trce env queryVar metrics latestPoints currentTip actionQueue =
       ClientStNext
         { recvMsgRollForward = \blk tip ->
               logException trce "recvMsgRollForward: " $ do
-                Gauge.set (withOrigin 0 (fromIntegral . unBlockNo) (getTipBlockNo tip)) (mNodeHeight metrics)
-                details <- getSlotDetails trce env queryVar (getTipPoint tip) (cardanoBlockSlotNo blk)
-                newSize <- atomically $ do
-                            writeDbActionQueue actionQueue $ mkDbApply blk details
-                            lengthDbActionQueue actionQueue
-                Gauge.set (fromIntegral newSize) $ mQueuePostWrite metrics
+                if withOrigin 0 unBlockNo (getTipBlockNo tip) == 0
+                  then do
+                    -- https://github.com/input-output-hk/cardano-db-sync/issues/433
+                    liftIO $ logWarning trce "Node has not yet received blocks from the network. Sleeping for 1 minute."
+                    threadDelay (60 * 1000 * 1000)
```

Comment:
This doesn't make any sense at all. We have been given an actual block `blk` above. Yet we're going to skip it and not add it to the queue?

It _might_ not break anything to skip the block entirely, but only only because it's likely that this block is the EBB at the beginning of the Byron chain, and as such contains nothing of interest.

It's not guaranteed, since you're looking at the tip of the chain, not the block itself, to make this decision.

Reply:
See https://github.com/input-output-hk/cardano-db-sync/issues/433#issuecomment-744144846